assets/js/base/components/cart-checkout/address-form/address-form.tsx

```diff
@@ -218,6 +241,7 @@ const AddressForm = ( {
 					<ValidatedTextInput
 						key={ field.key }
 						id={ `${ id }-${ field.key }` }
+						fieldName={ field.key }
```

Comment:
I have one idea to avoid passing in the field name to the component. What do you think of making `customValidationHandler` a curried function? So you pass something like `customValidationHandler( field.key )` and it returns a function?

Not required, just thought it might make it a tidier solution :)

Reply:
That's what I suggested [here](https://a8c.slack.com/archives/C8X6Q7XQU/p1677232444086039?thread_ts=1677219687.221979&cid=C8X6Q7XQU) I can't recall the tradeoffs thou

Reply:
It took me a bit to understand the intermediate function. I just updated the PR based on your suggestions, @mikejolley and @senadir.